Triplanar mapping node

Alpistinho opened the pull request:
I've added support for the triplanar mapping (https://github.com/LuxCoreRender/LuxCore/pull/252)
There is also some fixes for issues that I've found with my last pull request

![image](https://user-images.githubusercontent.com/7039667/67915144-5d6e2b00-fb4f-11e9-936a-930336a253c7.png)

Alpistinho commented:
That is the result from this node tree
![rock](https://user-images.githubusercontent.com/7039667/67915232-99a18b80-fb4f-11e9-8d44-0661208f41f2.png)

Alpistinho commented on nodes/materials/tree.py:
```diff
@@ -75,6 +75,7 @@ def poll(cls, context):
     LuxCoreNodeCategoryMaterial("LUXCORE_MATERIAL_TEXTURE", "Texture", items=[
         NodeItemMultiImageImport(),
         NodeItem("LuxCoreNodeTexImagemap", label="Image"),
+        NodeItem("LuxCoreNodeTriplanar", label="Triplanar mapping"),
```
Oops, some mismatched capitalization here

Theverat commented on export/aovs.py:
```diff
@@ -132,6 +132,9 @@ def convert(exporter, scene, context=None, engine=None):
 
+        if (use_adaptive_sampling and final):
+            props.Set(pyluxcore.Property("film.noiseestimation.index", adaptive_sampling_film_pipeline_index))
+
```
I would undo the changes in this file, my fixes were enough in my opinion.
I moved this code to line 130.

Theverat commented on nodes/textures/triplanar.py:
```diff
@@ -0,0 +1,24 @@
+    def init(self, context):
+        self.add_input("LuxCoreSocketColor", "Texture 1", [0.8, 0.0, 0.0])
+        self.add_input("LuxCoreSocketColor", "Texture 2", [0.0, 0.8, 0.0])
+        self.add_input("LuxCoreSocketColor", "Texture 3", [0.0, 0.0, 0.8])
```
Since these textures correspond to coordinate axis X, Y, Z, maybe we should name them as such so the user knows which texture is on which axis?
By the way, are they aligned to the local (object) or world coordinate system?

Dade916 replied:
I suggested @Alpistinho, in LuxCore pull request, to use a TextureMapping3D (GlobalMapping3D or LocalMapping3D) to map the hit point so the user can manipulate the trilinear texture as he wants.

Theverat replied:
Ok, so the node is missing a "LuxCoreSocketMapping3D" input.

You can look at another node how to add this, for example here:
https://github.com/LuxCoreRender/BlendLuxCore/blob/7f15ec345c1df215e1389b05bda9ae28663bb608/nodes/textures/blenderclouds.py#L26
And here it's exported:
https://github.com/LuxCoreRender/BlendLuxCore/blob/7f15ec345c1df215e1389b05bda9ae28663bb608/nodes/textures/blenderclouds.py#L42
https://github.com/LuxCoreRender/BlendLuxCore/blob/7f15ec345c1df215e1389b05bda9ae28663bb608/nodes/textures/blenderclouds.py#L53

Theverat commented:
There are still 3 things to fix for this project on the BlendLuxCore side, see the conversation items above (fix capitalization in menu, remove changes in aovs.py, add uvmapping3d socket).
@Alpistinho in case you don't have time, I can do this stuff in a few days.

Alpistinho commented:
Sorry, I've got the flu and couldn't look at this earlier.

I've reverted the AOV change and fixed the capitalization (Mapping is uppercased all around)
About the socket, the code is already there, no?

Theverat commented:
That's unfortunate, I hope you get well soon!
The pull request looks good now, thanks for the changes.
